src/plugins/intel_cpu/src/node.cpp

```diff
@@ -320,9 +320,37 @@ void Node::selectPreferPrimitiveDescriptor(const std::vector<impl_desc_type>& pr
                     auto parentDesc = parent_spd->getConfig().outConfs[inNum].getMemDesc();
 
                     const bool isCompatible = curDesc->isCompatible(*parentDesc);
-
-                    if (isCompatible) {
-                        equalsLocalFormatCount++;
+                    // Select Alogrithm:
+                    // Condition 1: Precision and shape keep same.
+                    // Condition 2: Static shape.
+                    // SShape=Shape(all elements = 1 or only 1 position is not 1)
+                    // 1: Compatible + not SShape:  score=100
+                    // 2: Compatible + SShape:      score=2
+                    // 3: Not Compatible + SShape:  score=1
+                    // 4: Others:                   score=0
+                    if ((curDesc->getPrecision() == parentDesc->getPrecision()) && curDesc->getShape().isStatic() &&
+                        curDesc->getShape() == parentDesc->getShape()) {
```

Comment:
If the current implementation is supposed to handle only static shapes, we better to check that parentDesc shape is static as well.

Reply:
Because we can't estimate the reoder calculate metric for dynamic shape. So I think it is only reasonable for static shape .